### ISSUE TITLE
Problems with catalog entity id in Magento 2.3.1 with Elasticsearch 6.7.1

### ISSUE THREAD

**Issue by Hanhan1989** [magento/magento2#22264]
### Preconditions (*)

1. Magento 2.3.1
2. Elasticsearch 6.7.1

### Steps to reproduce (*)

1. bin/magento setup:di:compile
2. bin/magento cache:clean
3. bin/magento indexer:reindex
1. Enter words in the search box and click search

### Expected result (*)

1. The searched products must appear

### Actual result (*)

1. Error: 

```
`1 exception(s):
Exception #0 (Magento\Framework\Exception\NoSuchEntityException): No such entity with id = 352

Exception #0 (Magento\Framework\Exception\NoSuchEntityException): No such entity with id = 352
#1 Magento\Catalog\Model\CategoryRepository->get(352) called at [vendor/magento/module-catalog/Model/CategoryList.php:82]
#2 Magento\Catalog\Model\CategoryList->getList(&Magento\Framework\Api\SearchCriteria#000000004a0722ed0000000056bf65bc#) called at [vendor/magento/module-elasticsearch/Model/Adapter/FieldMapper/Product/FieldProvider/DynamicField.php:102]
#3 Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\FieldProvider\DynamicField->getFields(array()) called at [vendor/magento/module-elasticsearch/Model/Adapter/FieldMapper/Product/CompositeFieldProvider.php:46]
#4 Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\CompositeFieldProvider->getFields() called at [vendor/magento/module-elasticsearch-6/Model/DataProvider/Suggestions.php:228]
#5 Magento\Elasticsearch6\Model\DataProvider\Suggestions->getSuggestFields() called at [vendor/magento/module-elasticsearch-6/Model/DataProvider/Suggestions.php:199]
#6 Magento\Elasticsearch6\Model\DataProvider\Suggestions->addSuggestFields(array('index'  [vendor/magento/module-advanced-search/Model/SuggestedQueries.php:66]
#9 Magento\AdvancedSearch\Model\SuggestedQueries->getItems(&Magento\Search\Model\Query#000000004a072cea0000000056bf65bc#) called at  .......`

```

**Comment by Hanhan1989** [comment]
I found a solution for this error by now. The solution is disable and enable again the fields 

* flat catalog category
* flat catalog product

![image](https://user-images.githubusercontent.com/11807165/55871359-c29a0900-5b8a-11e9-8ac6-5d7eb14b8eac.png)

But  the error appears again if we execute:

* bin/magento setup:di:compile
* bin/magento cache:clean
* bin/magento indexer:reindex

Is this a bug of 2.3.1 ?

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99143` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magepro-dev** [comment]
i have the same problem at magento 2.3.1. disabled FLAT categories and products.. and looks like it works now. but i hope in next release it will fixed

**Comment by Hanhan1989** [comment]
Any new news about this???

**Comment by magento-engcom-team** [comment]
Hi @Hanhan1989. 

Thank you for your report and collaboration!

The related internal Jira ticket `MAGETWO-99143` was closed as `non-reproducible` in `2.3-develop`.
It means that Magento team either unable to reproduce this issue using provided _Steps to Reproduce_ from the **_Description section_** on clean Magento instance or the issue has been already fixed in the scope of other tasks.

But if you still run into this problem please update or provide additional information/steps/preconditions in the **_Description section_** and reopen this issue.

**Comment by chris-pook** [comment]
Hi, just a quick heads up on this latest advice from Magento: 

> Starting with Magento 2.3.0+, the use of a flat catalog is no longer a best practice and is not recommended. Continued use of this feature is known to cause performance degradation and other indexing issues. A detailed description and solution is available in the Help Center.

https://docs.magento.com/m2/ce/user_guide/catalog/catalog-flat.html

This documentation isn't really clear on the reasoning and (also contradicts elsewhere in the same docs) but I have disabled flat catalog on sites now and it has fixed numerous issues.
